plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -262,7 +264,8 @@ protected SchemaTableName createMaterializedViewStorageTable(ConnectorSession se
         Map<String, Object> storageTableProperties = new HashMap<>(definition.getProperties());
         storageTableProperties.putIfAbsent(FILE_FORMAT_PROPERTY, DEFAULT_FILE_FORMAT_DEFAULT);
 
-        SchemaTableName storageTable = new SchemaTableName(viewName.getSchemaName(), storageTableName);
+        String storageSchema = getStorageSchema(definition.getProperties()).orElse(viewName.getSchemaName());
+        SchemaTableName storageTable = new SchemaTableName(storageSchema, storageTableName);
```

Comment:
Who validates that storageSchema does exist?
(previously it was kind of guaranteed by the engine)

Reply:
We leave it to the user/admin to ensure that the storage schema specified in MV property or catalog config exists. I've added a test with non existent schema to `testStorageSchemaProperty` that verifies the query will fail with Schema not found in exception message.